synapse/models/transport.py

```diff
@@ -52,17 +106,177 @@ def getModelDefs(self):
+                    'template': {
+                        'place': 'place',       # airport, seaport, starport
+                        'point': 'point',       # gate, slip, stargate...
+                        'vehicle': 'vehicle',   # air craft, vessel, space ship...
+                        'trip': 'trip'},        # flight, voyage...
+
+                    'props': (
+
+                        ('duration', ('duration', {}), {
+                            'doc': 'The actual duration.'}),
+
+                        ('departed', ('time', {}), {
+                            'doc': 'The actual departure time.'}),
+
+                        ('departed:place', ('geo:place', {}), {
+                            'doc': 'The actual departure {place}.'}),
+
+                        ('departed:point', ('transport:point', {}), {
+                            'doc': 'The actual departure {point}.'}),
+
+                        ('arrived', ('time', {}), {
+                            'doc': 'The actual arrival time.'}),
+
+                        ('arrived:place', ('geo:place', {}), {
+                            'doc': 'The actual arrival {place}.'}),
+
+                        ('arrived:point', ('transport:point', {}), {
+                            'doc': 'The actual arrival {point}.'}),
+
+                        ('scheduled:duration', ('duration', {}), {
+                            'doc': 'The scheduled duration.'}),
+
+                        ('scheduled:departure', ('time', {}), {
+                            'doc': 'The scheduled departure time.'}),
+
+                        ('scheduled:departure:place', ('geo:place', {}), {
+                            'doc': 'The scheduled departure {place}.'}),
+
+                        ('scheduled:departure:point', ('transport:point', {}), {
+                            'doc': 'The scheduled departure {point}.'}),
+
+                        ('scheduled:arrival', ('time', {}), {
+                            'doc': 'The scheduled arrival time.'}),
+
+                        ('scheduled:arrival:place', ('geo:place', {}), {
+                            'doc': 'The scheduled departure {place}.'}),
+
+                        ('scheduled:arrival:point', ('transport:point', {}), {
+                            'doc': 'The scheduled arrival {point}.'}),
+                    ),
+                }),
+
+                ('transport:trip', {
+                    # train, flight, drive, launch...
+                    'doc': 'Properties common to a specific trip taken by a vehicle.',
+                    'interfaces': ('transport:schedule',),
+
+                    'props': (
+
+                        ('status', ('str', {'enums': 'scheduled,cancelled,in-progress,completed,aborted,failed,unknown'}), {
+                            'doc': 'The status of the {trip}.'}),
+
+                        ('occupants', ('int', {'min': 0}), {
+                            'doc': 'The number of occupants of the {vehicle} on this {trip}.'}),
```

Comment:
Is there a way to capture how many occupants there were supposed to be vs how many occupants were actually on the trip. I'm thinking about a flight where there are a bunch of no-shows.

Reply:
Yes, the vehicle interface contains max occupants vs the trip's occupants. :+1:

Reply:
I'm still not sure that would answer the question. Say I have a 747 that has a maximum passenger count of 450. I have 300 scheduled passengers but only 295 of them show up. Is that something that can be modeled with this form? (Is it even something we need to consider? Again, modeling passenger flights seems like the primary use case that I can think of although maybe passenger trains too.)